caddyhttp/httpserver/plugin.go

```diff
@@ -340,6 +343,15 @@ func standardizeAddress(str string) (Address, error) {
 		}
 	}
 
+	// convert non-ascii (IDNA) host to Punycode
+	if isIDN(host) {
```

Comment:
What's the significance of this check? AFAIK converting an ASCII name like `example.com` will remain unchanged. Is the extra function, this if statement, etc etc, all necessary?